### PR TITLE
GA4 client_id

### PR DESCRIPTION
**Pull request by ishachopra0401**
## What do these changes do?

Updated doc with GA4 client_id information

### CODE REVIEW

File: docs/destinations/streaming-destinations/google-analytics-4/google-analytics-4-cloud-mode.mdx [hunk 1]
```diff
@@ -34,13 +34,24 @@ The mappings for the above-mentioned mandatory parameters are listed in the foll
...
 
+### Mapping `client_id` 
+
+RudderStack maps the `client_id` in the following order of precedence:
+
+- If a value is provided in the RudderStack dashboard's **GA4 client_id Field Identifier** setting, it looks up that value in the event payload and maps it to the GA4 `client_id`.
+- If no value is provided in the above dashboard setting or a value is provided in the dashboard but not in the payload, RudderStack maps the `client_id` from the following fields in the same priority order as listed below:
+  - From `externalID` `ga4ClientId`
+  - From `externalID` `anonymousId`
+  - From  `context.device.id`
```
Comment (mihir-4116) [inline docs/destinations/streaming-destinations/google-analytics-4/google-analytics-4-cloud-mode.mdx:52]: Remove - From `context.device.id`
this is not required now

**Review by mihir-4116 — CHANGES_REQUESTED** [review]

**Review by mihir-4116 — APPROVED** [review]

File: docs/destinations/streaming-destinations/google-analytics-4/google-analytics-4-cloud-mode.mdx [hunk 2]
```diff
@@ -34,13 +34,23 @@ The mappings for the above-mentioned mandatory parameters are listed in the foll
...
+| `client_id`         |  Refer to the <Link to="#mapping-client_id">Mapping `client_id`</Link> section. |
 | `app_instance_id`   |  from `externalID` `ga4AppInstanceId`  | 
 
 <div class="infoBlock">
 Refer to the <Link to="#how-do-i-obtain-app_instance_id">FAQ</Link> section for more information on how to obtain the <code class="inline-code">app_instance_id</code>.
 </div>
 
+### Mapping `client_id` 
+
+RudderStack maps the `client_id` in the following order of precedence:
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/google-analytics-4/google-analytics-4-cloud-mode.mdx:46]:
```suggestion
RudderStack maps the GA4 `client_id` in the following order of precedence:
```

File: docs/destinations/streaming-destinations/google-analytics-4/google-analytics-4-cloud-mode.mdx [hunk 3]
```diff
@@ -34,13 +34,23 @@ The mappings for the above-mentioned mandatory parameters are listed in the foll
...
+
+- If a value is provided in the RudderStack dashboard's **GA4 client_id Field Identifier** setting, it looks up that value in the event payload and maps it to the GA4 `client_id`.
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/google-analytics-4/google-analytics-4-cloud-mode.mdx:48]:
```suggestion
- If a value is provided in the RudderStack dashboard's **GA4 client_id Field Identifier** setting, it looks up that value in the event payload and maps it to `client_id`.
```

File: docs/destinations/streaming-destinations/google-analytics-4/google-analytics-4-cloud-mode.mdx [hunk 4]
```diff
@@ -34,13 +34,23 @@ The mappings for the above-mentioned mandatory parameters are listed in the foll
...
+- If no value is provided in the above dashboard setting or a value is provided in the dashboard but not in the payload, RudderStack maps the `client_id` from the following fields in the same priority order as listed below:
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/google-analytics-4/google-analytics-4-cloud-mode.mdx:49]:
```suggestion
- If no value is provided in the above dashboard setting or a value is provided in the dashboard but not in the event payload, then RudderStack maps `client_id` from the following fields in the exact priority order as listed below:
```

File: docs/destinations/streaming-destinations/google-analytics-4/google-analytics-4-cloud-mode.mdx [hunk 5]
```diff
@@ -34,13 +34,23 @@ The mappings for the above-mentioned mandatory parameters are listed in the foll
...
+  - From `externalID` `ga4ClientId`
+  - From `externalID` `anonymousId`
+  - From `messageId`
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/google-analytics-4/google-analytics-4-cloud-mode.mdx:52]: Wondering if it we should present this as 
`externalId: ga4ClientId`
`externalId: anonymousId` 

as `externalId` is an object within which `ga4ClientId` and `anonymousId` are the fields. 

Please correct me if I'm wrong here. @ishachopra0401 @mihir-4116

Comment (mihir-4116) [reply docs/destinations/streaming-destinations/google-analytics-4/google-analytics-4-cloud-mode.mdx:52]: no,`anonymousId`  will not come from externalId, just `anonymousId` is enough

**Review by mihir-4116 — APPROVED** [review]

**Review by ameypv-rudder — APPROVED** [review]
LGTM

**Pull request merged**